File: scala/src/main/scala/chapter01/Student.scala
package chapter01

class Student(val name: String, var age: Int) {
  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
  }

}

// 伴生对象
object Student {
  val school: String = "mit"

  def main(args: Array[String]): Unit = {
    new Student("scq", 34).printInfo()
  }
}

File: scala/src/main/java/com/kim/Student.java
package com.kim;

/**
 * Hello world!
 */
public class Student {

    private String name;

    private Integer age;

    private static String school = "mit";

    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo() {
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }

    public static void main(String[] args) {
        new Student("scq", 34).printInfo();
    }
}
